File: Lab/Combinador.c
#include <stdio.h>
#include <string.h>

int main(){

    char str1[50];
    char str2[50];
    char juncao[100];

    while (scanf("%s %s", str1, str2) != EOF){
        int tam1 = strlen(str1);
        int tam2 = strlen(str2);

        int maior = tam1 > tam2 ? tam1 : tam2;

        int pos = 0;

        for (int i = 0; i < maior; i++) {
            if (i < tam1){
                juncao[pos] = str1[i];
                pos++;
            }
            if (i < tam2){
                juncao[pos] = str2[i];
                pos++;
            }
        }

        juncao[pos] = '\0';

        printf("%s\n", juncao);
    }
    return 0;
}

File: Lab/Combinador.java
import java.util.Scanner;

public class Combinador{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while(sc.hasNext()){
            String str1 = sc.next();
            String str2 = sc.next();
            
            int tam1 = str1.length();
            int tam2 = str2.length();

            int maior = tam1 > tam2 ? tam1 : tam2;

            StringBuilder str = new StringBuilder();

            for(int i = 0; i < maior; i++){
                if(i < tam1){
                    str.append(str1.charAt(i));
                }
                if(i < tam2){
                    str.append(str2.charAt(i));
                }
            }

            System.out.println(str.toString());
        }
        sc.close();
    }
}